srcpkgs/libdpp/template

```diff
@@ -0,0 +1,46 @@
+# Template file for 'libdpp'
+pkgname=libdpp
+version=10.0.10
+revision=1
+wrksrc=DPP-${version}
+build_style=cmake
+cmake_builddir=build
+configure_args="-DCMAKE_INSTALL_PREFIX=/usr"
+hostmakedepends="pkg-config"
+makedepends="libsodium-devel opus-devel zlib-devel openssl-devel fmt-devel json-c++"
+short_desc="C++ Discord API Library for Bots"
+maintainer="Bo-Ru Ju <school.shsps@gmail.com>"
+license=Apache-2.0
+homepage="https://dpp.dev"
+distfiles="https://github.com/brainboxdotcc/DPP/archive/refs/tags/v${version}.tar.gz"
+checksum=2a1c26f606298e5b683d1e140219c434e61c4b22e8510fa2a2d5f7b6758dff95
+
+post_extract() {
+	find ${wrksrc} \( -iname "*.h" -o -iname "*.cpp" \) -exec sed -i -E "s/dpp\/(nlohmann|fmt)\//\1\//" '{}' \+
+	sed -i -E "s/install\(TARGETS dpp LIBRARY DESTINATION (.*)\)/install\(TARGETS dpp EXPORT dpp-targets LIBRARY DESTINATION \1\)\ninstall\(EXPORT dpp-targets DESTINATION \1\)/" CMakeLists.txt
+	sed -i -E "s/libdpp/dpp/" "${wrksrc}/CMakeLists.txt" "cmake/libdpp-config.cmake"
+	sed -i -E "s/libdpp-config/dpp-config/" "${wrksrc}/cmake/CPackSetup.cmake"
+	mv "${wrksrc}/cmake/libdpp-config.cmake" "${wrksrc}/cmake/dpp-config.cmake"
+	rm -rf "${wrksrc}/include/dpp/fmt"
+	rm -rf "${wrksrc}/include/dpp/nlohmann"
+	sed -i -E 's/"\$\{PROJECT_SOURCE_DIR\}\/include\/dpp\/nlohmann\/json.hpp"/\<nlohmann\/json.hpp\>/g' ${wrksrc}/CMakeLists.txt
+	sed -i -E 's/"\$\{PROJECT_SOURCE_DIR\}\/include\/dpp\/nlohmann\/json_fwd.hpp"/\<nlohmann\/json_fwd.hpp\>/g' ${wrksrc}/CMakeLists.txt
+}
+
+do_clean() {
+	rm -rf usr/lib/dpp-${version%.*}
```

Comment:
why is this necessary?

Reply:
Because it installs in both usr/lib/dpp and usr/lib/dpp-10.0.

Reply:
If it's in `masterdir/builddir/$wrksrc` or `masterdir/destdir/$pkgname-$version`, it should automatically be cleaned. Also, this function is only run if `xbps-src clean` is run